File: 1684 Count the Number of Consistent Strings/1684.py
class Solution:
    def update_valid(self, allowed):
        self.valid = [False] * 26
        for c in allowed:
            self.valid[ord(c) - ord('a')] = True
    
    def is_consistent(self, word):
        for c in word:
            if not self.valid[ord(c) - ord('a')]:
                return False
        return True

    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        self.update_valid(allowed)
        ans = 0
        for word in words:
            if self.is_consistent(word):
                ans += 1
        return ans
            

File: 1684 Count the Number of Consistent Strings/1684.cpp
class Solution {
    vector<bool> valid;

    void updateValid(string allowed){
        valid = vector<bool>(26, false);
        for (int i = 0; i<valid.size(); i++){
            valid[i] = false;
        }
        for (char c: allowed){
            valid[c - 'a'] = true;
        }
    }

    bool isConsistent(string word){
        for (char c: word){
            if (!valid[c - 'a']){
                return false;
            }
        }
        return true;
    }

public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        updateValid(allowed);
        int ans = 0;
        for (string word: words){
            if (isConsistent(word)){
                ans++;
            }
        }
        return ans;
    }
};
